pkg/drivers/kic/node/spec.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/minikube/pkg/drivers/kic/oci"
+	"k8s.io/minikube/pkg/minikube/command"
+)
+
+// Spec describes a node to create purely from the container aspect
+// this does not inlude eg starting kubernetes (see actions for that)
+type Spec struct {
+	Name              string
+	Profile           string
+	Role              string
+	Image             string // for example  4000mb based on https://docs.docker.com/config/containers/resource_constraints/
+	CPUs              string // for example 2
```

Comment:
Please don't use string for non-string values, unless you have to match an existing interface.